ria_remote/create_sibling_ria.py

```diff
@@ -0,0 +1,305 @@
+        #         protocol to send PREPARE)
+        #       - Alternatively we can create the remote directory and ria version file directly, but this means code
+        #         duplication that then needs to be kept in sync with ria-remote implementation.
+        #       - this leads to the third option: Have that creation routine importable and callable from
+        #         ria-remote package without the need to actually instantiate a RIARemote object
+        lgr.debug("initializing object store")
+        subprocess.run(['git', 'annex', 'fsck', '--from={}'.format(storage_sibling), '--fast', '--exclude=*/*'],
+                       cwd=str(ds.path))
+
+        # 2. create a bare repository in-store:
+        # determine layout locations
+        repo_path, archive_path, objects_path = RIARemote.get_layout_locations(base_path, ds.id)
+
+        lgr.debug("init bare repository")
+        # TODO: we should prob. check whether it's there already. How?
+        # Note: like the special remote itself, we assume local FS if no SSH host is specified
+        disabled_hook = sh_quote(str(repo_path / 'hooks' / 'post-update.sample'))
+        enabled_hook = sh_quote(str(repo_path / 'hooks' / 'post-update'))
+
+        if ssh_host:
+            from datalad import ssh_manager
+            ssh = ssh_manager.get_connection(ssh_host, use_remote_annex_bundle=False)
+            ssh.open()
+            ssh('cd {} && git init --bare'.format(sh_quote(str(repo_path))))
+            if not no_server:
+                ssh('mv {} {}'.format(disabled_hook, enabled_hook))
+        else:
+            cmd = ['git', 'init', '--bare']
+            subprocess.run(cmd, cwd=str(repo_path), check=True)
+            if not no_server:
+                cmd = ['mv', disabled_hook, enabled_hook]
+                subprocess.run(cmd, cwd=str(repo_path), check=True)
+
+        # add a git remote to the bare repository
+        # Note: needs annex-ignore! Otherwise we might push into default annex/object tree instead of
+        # directory type tree with dirhash lower. This in turn would be an issue, if we want to pack the entire thing
+        # into an archive. Special remote will then not be able to access content in the "wrong" place within the
+        # archive
+        lgr.debug("set up git remote")
+        # TODO: - This results in "[WARNING] Failed to determine if datastore carries annex."
+        #         (see https://github.com/datalad/datalad/issues/4028)
+        #       - additionally there's https://github.com/datalad/datalad/issues/3989, where datalad-siblings might
+        #         hang forever
+
+        if ssh_host:
+            lgr.error("XXXXXXXXXXXX\n%s", [str(p) for p in ssh_manager.socket_dir.iterdir()])
+            lgr.error("XXXXXXXXXXXX\n%s %s", ssh.ctrl_path, ssh.is_open())
+        ds.siblings(
+            'configure',
+            name=sibling,
+            url='{}:{}'.format(ssh_host, str(repo_path))
```

Comment:
`datalad.core.distributed.clone.decode_source_spec()` provides a `giturl` that can be used here. We should use this helper rather then going custom here.